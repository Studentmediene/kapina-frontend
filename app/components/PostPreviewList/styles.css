.postPreviewList {
  margin: 0 auto;
}

.post {
  margin-bottom: 1.5rem;
  padding-bottom: 1.5rem;
  border-bottom: 1px solid lightgray;
}
