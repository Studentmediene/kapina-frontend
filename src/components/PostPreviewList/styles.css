.postPreviewList {
  margin: 0 auto;
}

.post {
  border-bottom: 1px solid lightgray;
  margin-bottom: 1.5rem;
  padding-bottom: 1.5rem;
}
