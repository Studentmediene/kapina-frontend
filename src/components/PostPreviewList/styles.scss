@import 'variables.scss';

.postPreviewList {
  display: grid;
  grid-template-columns: 100%;
  grid-row-gap: 1.5rem;
}

.post {
  border-bottom: 1px solid lightgray;
  padding-bottom: 1.5rem;
}

@media screen and (min-width: $breakpoint-large) {
  .postPreviewList {
    grid-template-columns: 48% 48%;
    grid-column-gap: 4%;
  }

  .postLarge {
    grid-column-start: 1;
    grid-column-end: 3;
  }
}
