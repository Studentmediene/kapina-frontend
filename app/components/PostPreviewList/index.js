/**
*
* PostPreviewList
*
*/

import React from 'react';
import PostPreview from 'components/PostPreview';

import styles from './styles.css';

function PostPreviewList(props) {
  const posts = props.posts.map(
    (post, index) => (
      <div className={styles.post} key={`post-${index}`}>
        <PostPreview {...post} />
      </div>
    )
  );

  return (
    <div className={styles.postPreviewList}>
      {posts}
    </div>
  );
}

PostPreviewList.propTypes = {
  posts: React.PropTypes.array.isRequired,
};

export default PostPreviewList;
