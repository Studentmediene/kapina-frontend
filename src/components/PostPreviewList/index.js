import React from 'react';
import Immutable from 'immutable';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './styles.scss';
import PostPreview from 'components/PostPreview';
import HighlightedPosts from 'components/FrontPage/components/HighlightedPosts';
import HighlightedShows from 'components/FrontPage/components/HighlightedShows';

const PostPreviewList = props => {
  const posts = props.posts.toJS().map((post, index) => (
    <div
      className={classNames(styles.post, {
        [styles.postLarge]: index % 8 === 0 || index % 8 === 3,
      })}
      key={index}
    >
      <PostPreview {...post} />
    </div>
  ));
  posts.splice(
    2,
    0,
    <HighlightedShows
      //shows={{"test1", "test2", "test3"}}
      key={'highlightedShows'}
    />,
  );
  if (!props.highlightedPosts.isEmpty()) {
    // Insert highlighted posts after element 3
    posts.splice(
      3,
      0,
      <HighlightedPosts
        posts={props.highlightedPosts}
        key={'highlightedPosts'}
      />,
    );
  }

  return <div className={styles.postPreviewList}>{posts}</div>;
};

PostPreviewList.propTypes = {
  posts: PropTypes.instanceOf(Immutable.List).isRequired,
  highlightedPosts: PropTypes.instanceOf(Immutable.List),
};

export default PostPreviewList;
