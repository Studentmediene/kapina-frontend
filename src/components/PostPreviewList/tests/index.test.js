import React from 'react';
import { shallow } from 'enzyme';
import Immutable, { fromJS } from 'immutable'

import PostPreviewList from '../';

describe('<PostPreviewList />', () => {
  it('renders correctly with posts', () => {
    const mockProps = {
      posts: fromJS([
        {
          image: 'image',
          lead: 'Lead',
          publishAt: '2017-11-01 18:11:44+00:00',
          slug: 'slug1',
          title: 'post1',
          categories: [
            {
              backgroundColor: '#FFFFFF',
              name: 'categoryName1',
              textColor: '#000000',
            },
            {
              backgroundColor: '#FFFFFF',
              name: 'categoryName1',
              textColor: '#000000',
            },
          ],
        },
        {
          image: 'image',
          lead: 'Lead',
          publishAt: '2017-11-01 18:11:44+00:01',
          slug: 'slug2',
          title: 'post2',
          categories: [],
        },
      ]),
      highlightedPosts: Immutable.List(),
    };
    const tree = shallow(<PostPreviewList {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly without posts', () => {
    const mockProps = {
      posts: Immutable.List(),
      highlightedPosts: Immutable.List(),
    };
    const tree = shallow(<PostPreviewList {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});
