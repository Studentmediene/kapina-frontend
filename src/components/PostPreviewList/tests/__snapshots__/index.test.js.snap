// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostPreviewList /> renders correctly with posts 1`] = `
<div
  className="postPreviewList"
>
  <div
    className="post postLarge"
    key="0"
  >
    <PostPreview
      categories={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "name": "categoryName1",
            "textColor": "#000000",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "name": "categoryName1",
            "textColor": "#000000",
          },
        ]
      }
      image="image"
      lead="Lead"
      publishAt="2017-11-01 18:11:44+00:00"
      slug="slug1"
      title="post1"
    />
  </div>
  <div
    className="post"
    key="1"
  >
    <PostPreview
      categories={Array []}
      image="image"
      lead="Lead"
      publishAt="2017-11-01 18:11:44+00:01"
      slug="slug2"
      title="post2"
    />
  </div>
</div>
`;

exports[`<PostPreviewList /> renders correctly without posts 1`] = `
<div
  className="postPreviewList"
/>
`;
